From d18eaa1dce2390d811188eab06514af70acea3f7 Mon Sep 17 00:00:00 2001
From: Albin Tonnerre <albin.tonnerre@gmail.com>
Date: Fri, 19 Jun 2009 14:52:14 +0200
Subject: [PATCH 1/2] fix library soname

---
 configure.ac |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index 74e01a4..faeb558 100644
--- a/configure.ac
+++ b/configure.ac
@@ -12,6 +12,7 @@ AC_CANONICAL_HOST
 AC_ISC_POSIX
 
 AM_INIT_AUTOMAKE(1.6 dist-bzip2)
+AM_MAINTAINER_MODE([enable])
 AM_CONFIG_HEADER(config.h)
 
 AC_LIBTOOL_WIN32_DLL
@@ -26,7 +27,7 @@ SNAP=`echo $PACKAGE_VERSION | awk -F. '{printf("%s", $4);}'`
 version_info=`expr $VMAJ + $VMIN`":$VMIC:$VMIN"
 AC_SUBST(version_info)
 
-release="ver-svn-02"
+release="ver-pre-01"
 case "$host_os" in
    mingw32ce* | cegcc*)
       ;;
-- 
1.6.3.1

